src/pages/settings/subscriptions/common/hooks/useHandleChange.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Subscription } from 'common/interfaces/subscription';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { Dispatch, SetStateAction } from 'react';
+import { set } from 'lodash';
+
+interface Params {
+  setErrors: Dispatch<SetStateAction<ValidationBag | undefined>>;
+  setSubscription: Dispatch<SetStateAction<Subscription | undefined>>;
+  subscription: Subscription | undefined;
+}
+
+export function useHandleChange(params: Params) {
+  return (
+    property: keyof Subscription,
+    value: Subscription[keyof Subscription]
+  ) => {
+    params.setErrors(undefined);
+
+    const subscription = { ...params.subscription };
```

Comment:
This won't work as a deep clone if we have nested data, so `set` would cause an update by reference, which is probably what you don't want. 

Is data of `subscription` nested or one level (array/object)? If so, we can do a spread operator, if not make sure to deep clone. You can make use of `cloneDeep` from Lodash.

Reply:
I have a nested object, I just use the `set` fucntion for easier handling of nested properties like `webhook_configuration.post_purchase_url`. So I just changed this part to `cloneDeep`.